cellacdc/widgets.py

```diff
@@ -6754,6 +6762,39 @@ def pointsClicked(self, item, points, event):
         print('')
         print('*'*60)
 
+    def showIDs(self, annotate_labels_idxs=None, init=False):
+        if init:
+            self.annotate_labels_idxs = annotate_labels_idxs
+        if self.annotate_labels_idxs is None:
+            return
+        for i, plotItem in enumerate(self.PlotItems):
+            if i not in self.annotate_labels_idxs:
+                continue
+            imageItem = self.ImageItems[i]
+            lab = imageItem.image
+            rp = skimage.measure.regionprops(lab)
+            for obj in rp:
+                yc, xc = obj.centroid
+                textItem = pg.TextItem(
```

Comment:
Here I would store the `pg.TextItem` in a global `self.textItems` dictionary where the keys are the IDs of the objects. This way we do not need to recreate a TextItem and add it to the plot if it was already created previously. Creating and adding items is computationally expensive and there is no need to repeat it. We should also support 3D objects.